fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -98,9 +104,14 @@ def __init__(self,
                  grid_limit=2.0,
                  grid=None):
         self.estimator = estimator
-        if not isinstance(constraints, Moment):
-            raise RuntimeError("Unsupported disparity metric")
-        self.constraints = constraints
+        if isinstance(constraints, Moment):
+            self.constraints_ = constraints
```

Comment:
I think this trailing underscore is following `sklearn` - AIUI, the convention for this is `{ get: private set;}` in divers other languages

Reply:
It's generally used in python to avoid naming conflicts with keywords, https://stackoverflow.com/questions/21075208/whats-the-advantage-of-a-trailing-underscore-in-python-naming

Therefore I would not suggest using this. If you want to restrict get/set there are other ways of achieving that.

Reply:
I was going on the link from `sklearn` :
https://scikit-learn.org/stable/glossary.html#term-attributes
which states "A public attribute may have the same name as a constructor parameter, with a _ appended. This is used to store a validated or estimated version of the user’s input." Which is not quite what I wrote above :-)